bcda/auth/client/oktaclient.go

```diff
@@ -0,0 +1,128 @@
+package client
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/pborman/uuid"
+
+	"github.com/okta/okta-sdk-golang/okta"
+	"github.com/okta/okta-sdk-golang/okta/query"
+	"github.com/sirupsen/logrus"
+)
+
+var logger *logrus.Logger
+
+func init() {
+	logger = logrus.New()
+	logger.Formatter = &logrus.JSONFormatter{}
+
+	filePath, success := os.LookupEnv("BCDA_OKTA_LOG")
+	if success {
+		/* #nosec -- 0640 permissions required for Splunk ingestion */
+		file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0640)
+
+		if err == nil {
+			logger.SetOutput(file)
+		} else {
+			logger.Info("Failed to open Okta log file; using default stderr")
+		}
+	} else {
+		logger.Info("No Okta log location provided; using default stderr")
+	}
+
+	oktaToken := os.Getenv("OKTA_CLIENT_TOKEN")
+	if oktaToken == "" {
+		logger.Fatal("No Okta token found; please set OKTA_CLIENT_TOKEN")
+	}
+
+	oktaURL := os.Getenv("OKTA_CLIENT_ORGURL")
+	if oktaURL == "" {
+		logger.Fatal("No Okta URL found; please set OKTA_CLIENT_ORGURL")
+	}
+}
+
+func HealthCheck() (bool, error) {
+	reqId := uuid.NewRandom()
+	client := NewOktaClient()
+	oktaTestUser := "shawn@bcda.aco-group.us"
+
+	logRequest(reqId).Info("Okta ping request")
+	users, resp, err := client.User.ListUsers(query.NewQueryParams(query.WithQ(oktaTestUser)))
+	if err != nil {
+		logError(err, reqId).Error("Okta ping request error")
+		return false, err
+	}
+
+	if len(users) >= 1 {
+		logResponse(resp.StatusCode, reqId).Info("Okta ping request successful")
+		return true, err
+	}
+
+	logResponse(resp.StatusCode, reqId).Info("Okta ping request unsuccessful")
+	return false, nil
+}
+
+func DeleteUser(userId string) (bool, error) {
+	reqId := uuid.NewRandom()
+	client := NewOktaClient()
+	userField := logrus.Fields{"user_id": userId}
+
+	logRequest(reqId).WithFields(userField).Info("Okta delete user request")
+	resp, err := client.User.DeactivateUser(userId, nil)
+	if err != nil {
+		logError(err, reqId).WithFields(userField).Error("Okta delete user error")
+		return false, err
+	}
+
+	logResponse(resp.StatusCode, reqId).WithFields(userField).Info("Okta delete user success")
+	return true, err
+}
+
+func FindUser(email string) (string, error) {
+	reqId := uuid.NewRandom()
+	client := NewOktaClient()
+	emailField := logrus.Fields{"email": email}
```

Comment:
we probably shouldn't log this one (PII concerns).  i wrote up a simple tech spec on what i thought we should and should not be logging: https://confluence.cms.gov/display/BCDA/Tech+Spec%3A+Logging+for+Okta

but, i think that will be ironed out more in https://jira.cms.gov/browse/BCDA-866 if not here.